docs/user_guide/Upgrade_SIMP/Version_Maps/6.3.3_6.4.0.inc

```diff
@@ -71,3 +79,152 @@ functionality:
    for x in *.rpmsave; do
      mv $x `basename $x .rpmsave`
    done
+
+Upgrade Steps
+^^^^^^^^^^^^^
+These steps for upgrading are for use in SIMP systems that are not currently using
+r10k or Code Manager.
+
+Update OS and SIMP RPMS
+""""""""""""""""""""""""
+
+Update the repos for the OS and SIMP and run yum update
+
+The following errors/warnings are expected:
+
+.. code-block:: bash
+
+  warning: /var/simp/environments/simp/FakeCA/togen saved as /var/simp/environments/simp/FakeCA/togen.rpmsave
+  warning: /var/simp/environments/simp/FakeCA/ca.cnf saved as /var/simp/environments/simp/FakeCA/ca.cnf.rpmsave
+  libsemanage.semanage_direct_remove_key: Unable to remove module simp-environment at priority 400. (No such file or directory).
+  /usr/sbin/semodule:  Failed!
+
+If you are using the ``FakeCA`` in the ``simp`` environment, follow the steps in the ``FakeCA`` section above to restore the needed files.
+
+The old selinux module, ``simp-environment``, had to be removed before installing the
+the new selinux module, ``simp-environment-rsync`` to avoid conflicts when the
+simp-environment-skeleton module is installed.  Therefore
+the error above is given when the simp-environment RPM tries to delete it.
+
+Create the Puppetfile
+""""""""""""""""""""""
+At this point in the upgrade the puppet modules have been installed as git repositories
+under ``/usr/share/simp/modules``.  Now the Puppetfile, used by r10k to deploy modules
+to the environment, needs to be created.
+
+The environment name in this example is ``myenv``.  Run the following commands to create a default Puppetfile.
+
+.. code-block:: bash
+
+  # Change to the top level puppet environment directory for your envinroment.
+  cd /etc/puppetlabs/code/environments/myenv
+  # Generate the main Puppetfile
+  simp puppetfile generate --skeleton > Puppetfile
```

Comment:
- Once https://github.com/simp/rubygem-simp-cli/pull/124 goes through, will use the new ``--skeleton-with-local myenv`` instead

- Manual steps to put in the fully qualified path to Puppetfile.simp in Puppetfile may be required if https://simp-project.atlassian.net/browse/SIMP-6670 is not solved.

Reply:
* New option was changed to `--skeleton --local-modules myenv`
* Fully qualified path to Puppetfile.simp is now taken care of when the skeleton Puppetfile is generated.

Reply:
`--skeleton --local-modules myenv` won't help, because on upgrade all the SIMP modules will be detected as local repos, sigh.  So, disregard this whole discussion.